docs/trace/links-based-sampler/README.md

```diff
@@ -0,0 +1,162 @@
+# Links Based Sampling: An Example
+
+Certain scenarios such as a producer consumer scenario can be modelled using
+"span links" to express causality between activities. An activity (span) in a trace
+can link to any number of activities in other traces. When using a Parent Based
+sampler, the sampling decision is made at the level of a single trace. This implies
+that the sampling decision across such linked traces is taken independently without
+any consideration to the links. This can result in incomplete information to reason
+about a system. Ideally, it would be desirable to sample all linked traces together.
+
+As one possible way to address this, this example shows how we can increase the
+likelihood of having complete traces across linked traces.
+
+## How does this sampling example work?
+
+We use a composite sampler that has:
+
+1. A parent based sampler (this is a probabilistic / unbiased sampler).
+2. A links based sampler (this is a non-probabilistic/biased sampler).
+
+This composite sampler first delegates to the parent based sampler. If the
+parent based sampler decides to sample, then the composite sampler decides
+to sample. However, if the parent based sampler decides to drop, the composite
+sampler delegates to the links based sampler. The links based sampler decides
+to sample if the activity has any linked activities and if at least ONE of those
+linked activities is sampled.
+
+## When should you consider such an option?  What are the tradeoffs?
+
+This may be a good option to consider if you want to get more complete traces
+across linked traces. However, there are a few tradeoffs to consider:
+
+- **Not guaranted to give consistent sampling in all situations**: This
+approach doesn't guarantee that you will get complete traces across linked
+traces in all situations.
```

Comment:
Just a small nit, but another case where one possibly doesn't get complete traces is when links are added to non-root spans.

In this case, a sampled link might cause the span to be sampled, although the parent span is not sampled. The result would be an incomplete trace.

Not sure if it's worth adding this though.